src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -64,31 +73,66 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     List<Entity> events =
         new ArrayList<Entity>(results.asList(FetchOptions.Builder.withDefaults()));
 
-    // get location
-    // filter by location and cutoff outside it
+    // Get location of user
+    String location = request.getParameter("location");
+    LatLng userLocation = getLatLng(location);
+
+    // Get distance between user and the location of all the events
+    for (Entity event : events) {
+      LatLng eventLocation =
+          getLatLng(
+              event.getProperty("streetAddress").toString()
+                  + " "
+                  + event.getProperty("city").toString()
+                  + " "
+                  + event.getProperty("state").toString());
+      int distance = getDistance(userLocation, eventLocation);
+      event.setProperty("distance", distance);
+    }
+
+    // Get location cutoff converted from mi to km
+    int cutoff =
+        Math.toIntExact(
+            Math.round(
+                Integer.parseInt(Utils.getParameter(request, "searchDistance", "")) * 1.609));
```

Comment:
Maybe use a class constant for the 1.609?

Reply:
Good idea!